stages/org.osbuild.insights-client.config

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python3
+
+import fileinput
+import os
+import sys
+
+import osbuild.api
+
+
+def main(tree, options):
+    insights_client_config = options.get("config", {})
+    proxy = insights_client_config.get("proxy")
+    path = options.get("path", "/etc/insights-client/insights-client.conf")
+    path = os.path.join(tree, path.lstrip("/"))
+
+    changes = {}
+    if proxy is not None:
+        changes["proxy"] = {"key": "proxy", "value": proxy}
+
+    if not os.path.exists(path):
+        with open(path, 'w', encoding="utf8"):
+            pass
+
+    # For each of the configured options, find the first non-commented out instance
+    # of the option and replace it (if necessary). If it does not already exist, append
+    # the option to the end of the file.
+    # Keys are case insensitive, values are not. Try to preserve the key and default to
+    # camel-case.
+    with fileinput.input(files=(f"{path}"), inplace=True) as f:
+        for line in f:
+            line_list = line.split()
+            if len(line_list) == 2:
+                key, current_value = line_list
+                entry = changes.pop(key.lower(), None)
+                if entry is not None and current_value != entry['value']:
+                    sys.stdout.write(f"{key}={entry['value']}\n")
+                    continue
+            sys.stdout.write(line)
+    with open(f"{path}", mode="a", encoding="utf8") as f:
```

Comment:
Is the `f"{path}"` there to stringify the path?  It shouldn't be an issue if it's a pathlike.
```suggestion
    with open(path, mode="a", encoding="utf8") as f:
```